tests/test_js_signing.py

```diff
@@ -0,0 +1,55 @@
+"""
+Various tests related to validation for automated signing.
+"""
+
+from js_helper import TestCase
+
+
+class TestSearchService(TestCase):
+    """Tests that warnings related to the search service trigger warnings."""
+
+    def test_changes(self):
+        """Tests that setting changes trigger warnings."""
+
+        def test(obj, prop, stuff, warnings):
+            self.setUp()
+            self.run_script("""%s.%s%s;""" % (obj, prop, stuff))
+            self.assert_failed(with_warnings=warnings)
+
+        objs = ("Cc[''].getService(Ci.nsIBrowserSearchService)",
+                "Services.search")
+
+        for obj in objs:
+            warnings = {"signing_severity": "high",
+                        "id": ("testcases_javascript_actions",
+                               "search_service",
+                               "changes")},
+
+            for prop in "currentEngine", "defaultEngine":
+                yield test, obj, prop, " = foo", warnings
+
+            warnings[0]["signing_severity"] = "medium"
+
+            for meth in ("addEngine", "addEngineWithDetails",
+                         "removeEngine", "moveEngine"):
+                yield test, obj, meth, "(foo, bar, baz)", warnings
+
+    def test_registry_write(self):
+        """Tests that Windows registry writes trigger warnings."""
+
+        warnings = ({"id": ("testcases_javascript_actions",
+                            "windows_registry", "write"),
+                     "signing_severity": "medium"},
+                    {"id": ("js", "traverser", "dangerous_global"),
+                     "signing_severity": "low"})
+
+        def test(method):
+            self.setUp()
+            self.run_script("""
+                Cc[""].createInstance(Ci.nsIWindowsRegKey).%s(foo, bar);
+            """ % method)
+            self.assert_failed(with_warnings=warnings)
+
+        for method in ("create", "createChild", "writeBinaryValue",
+                       "writeInt64Value", "writeIntValue", "writeStringValue"):
+            yield test, method
```

Comment:
Yes, but if it fails and short-circuits the rest of the tests, it's not always clear which parts are failing, which often means multiple runs after partial fixes to see what's going on. This is particularly true for the other test cases that check both `Services.*` and `.getService(*)`, where the failure could be in the definition on the `Services` object or in the definition of the interface. If all of the tests run regardless of one failure, it's much more obvious where the problem lies before you start digging into code.